### PR TITLE
Add docs for combinatorial tokens

### PR DESCRIPTION
**Pull request by maltekliemann**

### CODE REVIEW

File: docs/learn/combinatorial-tokens.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+redeemed for `x` units of `X` since the payout of `(A|B)` is equal to `1`. To
+take a more complex example, if we have a scalar market with tokens `L` and `S`
+(long and short) and a payout vector of `(.6, .4)` and we want to redeem `x`
+units of `L&(X|Y)`, then we receive `.6` times `x` units of `(X|Y)`.
+
+## Combinatorial Betting
+
+Buying individual outcomes of a combinatorial market allows the user to bet on
+more specialzied outcomes (A _and_ B will happen), but the true strength of
+combinatorial markets lies is making bets on contingencies (if A occurrs, then
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
combinatorial markets lies in making bets on contingencies (if A occurs, then
```

File: docs/learn/combinatorial-tokens.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# Combinatorial Betting
+
+Zeitgeist's combinatorial tokens allow users to combine outcome tokens of
+several markets and trade these combined tokens in a single liquidity pool.
+
+Much of Zeitgeist's implementation of combinatorial tokens is based on pioneer
+work done by Gnosis done
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
work done by Gnosis
```

File: docs/learn/combinatorial-tokens.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+## Splits and Merges
+
+Suppose we have two markets with outcome tokens `A`, `B`, `C` and `X`, `Y`, `Z`,
+resp. Their tokens can be combined to the following tokens: `A&X`, `A&Y`, `A&Z`,
+`B&X`, `B&Y`, `B&Z`, `C&X`, `C&Y`, `C&Z`. A bet on the token `A&X` is a bet that
+both `A` _and_ `X` occur (we'll interpret the meaning of `A&X` in case one or
+both markets is a scalar market below in the section _Redeeming_).
+
+In fact, we can even go further and combine positions from both markets into
+_split positions such as `(A|B)&Z`, which signifies that `A` or `B` \_and_ `Z`
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
_split positions_ such as `(A|B)&Z`, which signifies that `A` or `B` and `Z`
```

File: docs/learn/combinatorial-tokens.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+there is a market with outcomes `U`, `V`.
+
+Merging is the opposite process. For example, `x` units of `(A|B)&(X|Y)` and
+`(A|B)&Z` may be combined into `x` units of `(A|B)`.
+
+## Redeeming
+
+If a user owns a position involving multiple markets, such as `(A|B)&X` and only
+one of the markets is resolved (say, the market with `A`, `B`, `C`), then they
+may want cash out the part of the position. This is where payout vectors come
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
may want to cash out the part of the position. This is where payout vectors come
```

File: docs/learn/combinatorial-tokens.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+Returning to the idea of redeeming tokens, `x` units of `(A|B)&X` can be
+redeemed for `x` units of `X` since the payout of `(A|B)` is equal to `1`. To
+take a more complex example, if we have a scalar market with tokens `L` and `S`
+(long and short) and a payout vector of `(.6, .4)` and we want to redeem `x`
+units of `L&(X|Y)`, then we receive `.6` times `x` units of `(X|Y)`.
+
+## Combinatorial Betting
+
+Buying individual outcomes of a combinatorial market allows the user to bet on
+more specialzied outcomes (A _and_ B will happen), but the true strength of
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
more specialised outcomes (A _and_ B will happen), but the true strength of
```

File: docs/learn/combinatorial-tokens.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+sake of simplicity, we denote these tokens by $D$, $D \land H$ and
+$D \land \bar H$, resp. By combining the two markets above, we create a new
+market with four assets: $D \land H$, $D \land \bar H$, $\bar D \land H$ and
+$\bar D \land \bar H$.
+
+Using splits, agents can now bet on correlations between the events $D$ and $H$.
+For example, to bet that the Dems win if Hillary is nominated, the agent would
+split their collateral into $x$ units of $H$ and $\bar H$, then split these
+units into $x$ units of $H \land D$ and $H \land \bar D$, then sell their units
+of $H \land \bar D$ for more $H \land D$. There are three possible outcomes:
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
of $H \land \bar D$ to acquire more $H \land D$. There are three possible outcomes:
```

File: docs/learn/combinatorial-tokens.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+> change as little as possible besides $p(A|B)$ (and of course
+> $p(\bar A|B) = 1 − p(A|B)$).
+
+(The market maker implemented in zrml-neo-swaps has that property.)
+
+A fundamental idea is that trades made on prediction markets are not so much
+absolute bets ("I bet $10 that Hillary will win at 1:1 odds"), but rather that
+trades are made to "correct" the forecast of the market by changing the prices
+of the outcome. The agent executing the trade pays a price to do so, but is
+rewarded if the prediction is closer to the truth that the current one.
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
rewarded if the prediction is closer to the truth than the current one.
```

File: docs/learn/combinatorial-tokens.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+> $p(C) = \sum\_{i \in C} p_i$. It seems preferable, however, for this bet to
+> change as little as possible besides $p(A|B)$ (and of course
+> $p(\bar A|B) = 1 − p(A|B)$).
+
+(The market maker implemented in zrml-neo-swaps has that property.)
+
+A fundamental idea is that trades made on prediction markets are not so much
+absolute bets ("I bet $10 that Hillary will win at 1:1 odds"), but rather that
+trades are made to "correct" the forecast of the market by changing the prices
+of the outcome. The agent executing the trade pays a price to do so, but is
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
trades are made to "correct" the forecast of the market by changing the prices
of the outcomes. The agent executing the trade pays a price to do so, but is
```

File: docs/learn/combinatorial-tokens.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+how likely $H$ is to occur. This is summarized in the _modularity property_
+defined by Hanson in [H03a]:
+
+> Consider the case of an agent who bets with a market scoring rule, and bets
+> only on the event $A$ given the event $B$. That is, this agent gains assets of
+> the form “Pays \$1 if $A$ and $B$ hold”, in trade for assets of the form “Pays
+> \$1 if $B$ holds and $A$ does not. [...] In general, depending on the
+> particular market scoring rule, such a bet might change any probability
+> estimate $p_i$, and thus change any event probability
+> $p(C) = \sum\_{i \in C} p_i$. It seems preferable, however, for this bet to
```
Comment (Chralt98) [inline docs/learn/combinatorial-tokens.md]:
```suggestion
> $p(C) = \sum_{i \in C} p_i$. It seems preferable, however, for this bet to
```

File: docs/learn/combinatorial-tokens.md (hunk 5 above)
Comment (maltekliemann) [reply docs/learn/combinatorial-tokens.md]: 257d4cbb7102d4b8f6cdf366815abd7f2999454b

**Review by Chralt98 — APPROVED** [review]

**Pull request merged**